zivid_camera/src/zivid_camera.cpp

```diff
@@ -503,6 +522,12 @@ rcl_interfaces::msg::SetParametersResult ZividCamera::setParametersCallback(
     RCLCPP_DEBUG_STREAM(
       get_logger(), "Set parameter '" << param.get_name() << "' (" << param.get_type_name()
                                       << ") to '" << param.value_to_string() << "'");
+    if (settings_controller_) {
+      settings_controller_->onSetParameter(param.get_name());
+    }
+    if (settings_2d_controller_) {
+      settings_2d_controller_->onSetParameter(param.get_name());
+    }
```

Comment:
These variables should always be non-null? So we don't need the if-checks.

Reply:
There is an initialization order issue complicating these matters. The callback is set before the settings controllers are constructed, thus they will be null for the first few callbacks.

We could move the callback to the bottom, then the default parameters being set won't be logged. We could also initialize all of zivid before the ros things, but that might have its own problems, e.g. if the zivid application won't start because of a bad configuration.